util/src/main/java/tech/pegasys/teku/util/config/KeyStoreFilesLocator.java

```diff
@@ -0,0 +1,156 @@
+    this.pathMap = new HashMap<>();
+    this.pathSeparator = pathSeparator;
+  }
+
+  public void parse() {
+    for (final String currentEntry : colonSeparatedPairs) {
+      if (!currentEntry.contains(pathSeparator)) {
+        throw new InvalidConfigurationException(
+            "validatorKeys entry ("
+                + currentEntry
+                + ") did not contain key and password separated by '"
+                + pathSeparator
+                + "' as expected.");
+      }
+      if (currentEntry.matches(".*:.*:.*")) {
+        throw new InvalidConfigurationException(
+            "validatorKeys entry ("
+                + currentEntry
+                + ") contained more than one '"
+                + pathSeparator
+                + "', not keyFile'"
+                + pathSeparator
+                + "'passFile as expected.");
+      }
+      final List<String> entry = Splitter.on(pathSeparator).splitToList(currentEntry);
+      parseEntry(entry.get(0), entry.get(1));
+    }
+  }
+
+  public void parseKeyAndPasswordList(
+      final List<String> keystoreFiles, final List<String> keystorePasswordFiles) {
+    for (int i = 0; i < keystoreFiles.size(); i++) {
+      parseEntry(keystoreFiles.get(i), keystorePasswordFiles.get(i));
+    }
+  }
+
+  void parseEntry(final String keyFileName, final String passwordFileName) {
+    final File keyFile = new File(keyFileName);
+    final File passwordFile = new File(passwordFileName);
+
+    if (!keyFile.exists()) {
+      throw new InvalidConfigurationException(
+          String.format("Invalid configuration. could not find the key file (%s).", keyFileName));
+    }
+    if (!passwordFile.exists()) {
+      throw new InvalidConfigurationException(
+          String.format(
+              "Invalid configuration. could not find the password file (%s).", passwordFileName));
+    }
+    if (keyFile.isDirectory() != passwordFile.isDirectory()) {
+      throw new InvalidConfigurationException(
+          String.format(
+              "Invalid configuration. validatorKeys entry (%s"
+                  + pathSeparator
+                  + "%s) must be both directories or both files",
+              keyFileName,
+              passwordFileName));
+    }
+    if (keyFile.isFile()) {
+      pathMap.putIfAbsent(keyFile.toPath(), passwordFile.toPath());
+      filePairs.add(Pair.of(keyFile.toPath(), passwordFile.toPath()));
+    } else {
+      parseDirectory(keyFile, passwordFile);
+    }
+  }
+
+  void parseDirectory(final File keyDirectory, final File passwordDirectory) {
+    final String keyBasePath = keyDirectory.getAbsolutePath();
+    final String passwordBasePath = passwordDirectory.getAbsolutePath();
+    try (Stream<Path> walk = Files.walk(keyDirectory.toPath())) {
+      walk.filter(Files::isRegularFile)
+          .filter(
+              (path) ->
+                  !path.getFileName().toString().startsWith(".")
+                      && path.toString().endsWith(".json"))
```

Comment:
Minor thing but since we're going to need to manipulate the filename later I'd make this:
```suggestion
                      && getFileName().toString().endsWith(".json"))
```